src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilterTest.java

```diff
@@ -0,0 +1,604 @@
+
+        try {
+            verifyFilterWithInlineConfigParser(
+                getPath("InputSuppressWithNearbyTextFilterInvalidLineRange.txt"),
+                violationAndSuppressedMessages
+            );
+            assertWithMessage("CheckstyleException is expected").fail();
+        }
+        catch (CheckstyleException ex) {
+            assertWithMessage("Invalid exception message")
+                .that(ex)
+                .hasCauseThat()
+                .hasMessageThat()
+                .isEqualTo("unable to parse line range"
+                        + " from 'SUPPRESS CHECKSTYLE LineLengthCheck' using a!b");
+        }
+    }
+
+    @Test
+    public void testAcceptNullViolation() {
+        final SuppressWithNearbyTextFilter filter =
+                new SuppressWithNearbyTextFilter();
+        final AuditEvent auditEvent = new AuditEvent(this);
+        assertWithMessage("Filter should accept audit event")
+                .that(filter.accept(auditEvent))
+                .isTrue();
+        assertWithMessage("File name should not be null")
+            .that(auditEvent.getFileName())
+            .isNull();
+    }
+
+    @Test
+    public void testThrowsIllegalStateExceptionWhenFileNotFound() {
+        final Violation message = new Violation(1, 1, 1, TokenTypes.CLASS_DEF,
+            "messages.properties", "key", null, SeverityLevel.ERROR, null, getClass(), null);
+        final String fileName = "nonexisting_file";
+        final AuditEvent auditEvent = new AuditEvent(this, fileName, message);
+
+        final SuppressWithNearbyTextFilter filter =
+                new SuppressWithNearbyTextFilter();
+
+        try {
+            filter.accept(auditEvent);
+            assertWithMessage(IllegalStateException.class.getSimpleName() + " is expected").fail();
+        }
+        catch (IllegalStateException ex) {
+            assertWithMessage("Invalid exception message")
+                .that(ex.getMessage())
+                .isEqualTo("Cannot read source file: " + fileName);
+
+            final Throwable cause = ex.getCause();
+            assertWithMessage("Exception cause has invalid type")
+                    .that(cause)
+                    .isInstanceOf(FileNotFoundException.class);
+            assertWithMessage("Invalid exception message")
+                .that(cause)
+                .hasMessageThat()
+                .isEqualTo(fileName + " (No such file or directory)");
+        }
+    }
+
+    @Test
+    public void testFilterWithDirectory() throws IOException {
+        final SuppressWithNearbyTextFilter filter =
+                new SuppressWithNearbyTextFilter();
+        final AuditEvent event = new AuditEvent(this, getPath(""), new Violation(1, 1,
+                "bundle", "key", null, SeverityLevel.ERROR, "moduleId", getClass(),
+                "customMessage"));
+
+        assertWithMessage("filter should accept directory")
+                .that(filter.accept(event))
+                .isTrue();
```

Comment:
please put a comment on all test methods that are pure-unit-test (not execution of whole-checkstyle) and explain why it is not possible to use `verify....` like approach.
We try to discourage all to use pure-unit-test and preserve a reason in comment if it is good reason and not obvious.

Reply:
Done.